src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,389 @@
+
+/-- The sum of two vector measure is a vector measure. -/
+def add (v w : vector_measure α M) : vector_measure α M :=
+{ measure_of' := v + w,
+  empty' := by simp,
+  not_measurable' := λ _ hi,
+    by simp [v.not_measurable hi, w.not_measurable hi],
+  m_Union' := λ f hf₁ hf₂,
+    has_sum.add (v.m_Union hf₁ hf₂) (w.m_Union hf₁ hf₂) }
+
+instance : has_add (vector_measure α M) := ⟨add⟩
+
+@[simp] lemma coe_add (v w : vector_measure α M) : ⇑(v + w) = v + w := rfl
+lemma add_apply (v w : vector_measure α M) (i : set α) :(v + w) i = v i + w i := rfl
+
+instance : add_comm_monoid (vector_measure α M) :=
+function.injective.add_comm_monoid _ coe_injective coe_zero coe_add
+
+/-- `coe_fn` is an `add_monoid_hom`. -/
+@[simps]
+def coe_fn_add_monoid_hom : vector_measure α M →+ (set α → M) :=
+{ to_fun := coe_fn, map_zero' := coe_zero, map_add' := coe_add }
+
+end add_comm_monoid
+
+section add_comm_group
+
+variables {M : Type*} [add_comm_group M] [topological_space M]
+
+variables [topological_add_group M]
+
+/-- The negative of a vector measure is a vector measure. -/
+def neg (v : vector_measure α M) : vector_measure α M :=
+{ measure_of' := -v,
+  empty' := by simp,
+  not_measurable' := λ _ hi, by simp [v.not_measurable hi],
+  m_Union' := λ f hf₁ hf₂, has_sum.neg $ v.m_Union hf₁ hf₂ }
+
+instance : has_neg (vector_measure α M) := ⟨neg⟩
+
+@[simp] lemma coe_neg (v : vector_measure α M) : ⇑(-v) = - v := rfl
+lemma neg_apply (v : vector_measure α M) (i : set α) :(-v) i = - v i := rfl
+
+/-- The difference of two vector measure is a vector measure. -/
+def sub (v w : vector_measure α M) : vector_measure α M :=
+{ measure_of' := v - w,
+  empty' := by simp,
+  not_measurable' := λ _ hi,
+    by simp [v.not_measurable hi, w.not_measurable hi],
+  m_Union' := λ f hf₁ hf₂,
+    has_sum.sub (v.m_Union hf₁ hf₂)
+      (w.m_Union hf₁ hf₂) }
+
+instance : has_sub (vector_measure α M) := ⟨sub⟩
+
+@[simp] lemma coe_sub (v w : vector_measure α M) : ⇑(v - w) = v - w := rfl
+lemma sub_apply (v w : vector_measure α M) (i : set α) : (v - w) i = v i - w i := rfl
+
+instance : add_comm_group (vector_measure α M) :=
+function.injective.add_comm_group _ coe_injective coe_zero coe_add coe_neg coe_sub
+
+end add_comm_group
+
+end vector_measure
+
+namespace measure
+
+/-- Given two finite measures `μ, ν`, `sub_to_signed_measure μ ν` is the signed measure
+corresponding to the function `μ - ν`. -/
+def sub_to_signed_measure (μ ν : measure α) [hμ : finite_measure μ] [hν : finite_measure ν] :
+  signed_measure α :=
+μ.to_signed_measure + - ν.to_signed_measure
+
+lemma sub_to_signed_measure_apply {μ ν : measure α} [finite_measure μ] [finite_measure ν]
+  {i : set α} (hi : measurable_set i) :
+  μ.sub_to_signed_measure ν i = (μ i).to_real - (ν i).to_real :=
+begin
+  rw [sub_to_signed_measure, vector_measure.add_apply, vector_measure.neg_apply,
+      to_signed_measure_apply_measurable hi, measure.to_signed_measure_apply_measurable hi,
+      sub_eq_add_neg]
+end
```

Comment:
This seems quite odd to me - is it the case that `μ.to_signed_measure - ν.to_signed_measure = (μ - ν).to_signed_measure`?

Reply:
Oh right, the point is you can't subtract `measure`s?

Reply:
`μ - ν` is nonnegative while `μ.to_signed_measure - ν.to_signed_measure` can be negative so the rhs is zero whenever the lhs is negative